python/ray/serve/scripts.py

```diff
@@ -98,6 +98,17 @@ def process_dict_for_yaml_dump(data):
     return data
 
 
+def override_or_default(
```

Comment:
Could we replace this helper function altogether with [`setdefault()`](https://www.w3schools.com/python/ref_dictionary_setdefault.asp)? The equivalent logic would be:

```python
if override_value is not None:
    dict[key] = override_value
else:
    dict.setdefault(key, default_value)
```

Reply:
I've changed the helper to use `setdefault()`; did you want to get rid of the helper somehow?

Reply:
Yeah, we only use the helper in two places, and it's relatively short. I thought we could remove the helper altogether and use the implementation directly.

Reply:
Ah, makes sense. Changed!